sympy/matrices/matrices.py

```diff
@@ -1001,8 +1001,8 @@ def _handle_creation_inputs(cls, *args, **kwargs):
                 if evaluate:
                     def do(x):
                         # make Block and Symbol explicit
-                        if isinstance(x, (list, tuple)):
-                            return type(x)([do(i) for i in x])
+                        if is_sequence(x):
+                            return list([do(i) for i in x])
```

Comment:
`return [do(i) for i in x]` should be enough.

(If this should be changed or not, I cannot really tell though...)

Reply:
This comments should go to #21941